modules/data.atmosphere/R/met_temporal_downscale.Gaussian_ensemble.R

```diff
@@ -0,0 +1,398 @@
+      for (i in seq_along(1:12)) {
+        m[i] <- Hmisc::monthDays(as.Date(paste0(year, "-", i, "-01")))
+      }
+      bmlist <- vector()
+      
+      Bm <- c(0.2089, 0.2857, 0.2689, 0.2137, 0.1925, 0.2209, 0.2527, 0.2495, 
+              0.2232, 0.1728, 0.1424, 0.1422)
+      for (x in seq_along(Bm)) {
+        mlen <- list()
+        mlen <- rep(Bm[x], m[x] * 24/reso)
+        bmlist <- append(bmlist, mlen)
+      }
+      A <- 0.73
+      C <- 0.7
+      hdry <- vector()
+      for (i in seq_along(precip)) {
+        if (precip[i] > 0) {
+          p <- 0.65
+        }
+        if (precip[i] == 0) {
+          p <- 2
+        }
+        hdry[i] <- A * p * (1 - exp(-1 * bmlist[i] * ((temp_max[i] - temp_min[i])^C)))
+      }
+      hdry[hdry < 0] <- 0
+      swflux <- hdry * I
+      swflux[swflux < 0] <- 0
+    }
+    
+    
+    # Putting all the variables together in a data frame
+    df$surface_downwelling_shortwave_flux_in_air <- swflux
+    downscaled.met <- data.frame(df)
+    
+    train.list <- list()
+    lat <- ncdf4::ncdim_def(name = "latitude", units = "degree_north", vals = lat_train, 
+                            create_dimvar = TRUE)
+    lon <- ncdf4::ncdim_def(name = "longitude", units = "degree_east", vals = lon_train, 
+                            create_dimvar = TRUE)
+    time <- ncdf4::ncdim_def(name = "time", units = "sec", vals = (1:reso_len) * 
+                             reso * 3600, create_dimvar = TRUE, unlim = TRUE)
+    dim <- list(lat, lon, time)
+    
+    for (j in seq_along(var$CF.name)) {
+      train.list[[j]] <- ncdf4::ncvar_def(name = as.character(var$CF.name[j]), 
+                                          units = as.character(var$units[j]), dim = dim, missval = -999, verbose = verbose)
+    }
+    
+    rows <- 1
+    dir.create(outfolder, showWarnings = FALSE, recursive = TRUE)
+    results <- data.frame(file = character(rows), host = character(rows), mimetype = character(rows), 
+                          formatname = character(rows), startdate = character(rows), enddate = character(rows), 
+                          dbfile.name = paste0(source_name, ".dwnsc.ens", e, ".", year, sep = "."), 
+                          stringsAsFactors = FALSE)
+    
+    
+    loc.file <- file.path(outfolder, paste0(source_name, ".dwnsc.gauss.ens", 
+                                            e, ".", year, ".nc"))
+    
+    loc <- ncdf4::nc_create(filename = loc.file, vars = train.list, verbose = verbose)
+    for (j in seq_along(var$CF.name)) {
+      ncdf4::ncvar_put(nc = loc, varid = as.character(var$CF.name[j]), vals = downscaled.met[[j]])
+    }
+    ncdf4::nc_close(loc)
+    
+    results$file <- loc.file
+    results$host <- PEcAn.utils::fqdn()
+    results$startdate <- paste0(year, "-01-01 00:00:00", tz = "UTC")
+    results$enddate <- paste0(year, "-12-31 23:59:59", tz = "UTC")
+    results$mimetype <- "application/x-netcdf"
+    results$formatname <- "CF Meteorology"
+    
+    invisible(results)
```

Comment:
First, by placing the return here you will exit the downscaling after the first ensemble member -- you need to return the results outside the loop!
Second, as is done here https://github.com/PecanProject/pecan/blob/master/modules/data.atmosphere/R/download.PalEON_ENS.R#L45 you'll want to put all these results in a list of data frames (one per ensemble member) rather than a single data frame.